lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -56,6 +65,15 @@ defmodule Kernel.ParallelCompiler do
     spawn_compilers(files, path, options)
   end
 
+  @doc """
+  Requires the given files in parallel.
+  Read `files/2` for more information.
+  """
+  def require_files(files, options \\ []) do
```

Comment:
I wanted to add warning and error returns to `ParallelRequire` in order to return diagnostics from the test compiler for test files. I started adding it to `ParallelRequire` but it seemed a fair amount of code duplication. I handled the different semantics for `require` by not setting the custom error handler in `ParallelCompile` if the `require` option is set: https://github.com/elixir-lang/elixir/pull/6446/files#diff-59dfea30c24e10aaa6d3e55047e926d6R144

If you like, I'll remove the changes to `ParallelRequire` from this PR, but adding warning and error returns to `ParallelRequire` will further increase the code duplication if we don't unify them at some point.